### PR TITLE
OXT-1016: Only build the Xen recipes once per build

### PR DESCRIPTION
**Pull request by dozylynx**
Remove the MACHINE-specific dependencies affecting the Xen recipes so that they no longer have to be recompiled per MACHINE.

PRs for: xenclient-oe, openxt and meta-openxt-ocaml-platform.

Signed-off-by: Christopher Clark <christopher.clark6@baesystems.com>

### CODE REVIEW

**Comment by jandryuk** [comment]
I ran a quick build test this morning.  Without the associated PRs, running a xen-libxl build for dom0 followed by ndvm changes the package.
$ MACHINE=xenclient-dom0 ./bb xen-libxl
$ sha1sum tmp-glibc/deploy/ipk//xen-libxl
dcafde6f5aef1f22ff24d0e9f18b126bc372c2fb  tmp-glibc/deploy/ipk/core2-32/xen-libxl-dbg_4.6.4-r0_core2-32.ipk
$ MACHINE=xenclient-ndvm ./bb xen-libxl
$ sha1sum tmp-glibc/deploy/ipk//xen-libxl
496064741f6903fb0ef6a6792e5a043c1d1f59be  tmp-glibc/deploy/ipk/core2-32/xen-libxl-dbg_4.6.4-r0_core2-32.ipk

With the PRs applied, the dom0 build creates the package.
$ MACHINE=xenclient-dom0 ./bb xen-libxl
$ sha1sum tmp-glibc/deploy/ipk//xen-libxl
474141cbc20325c910679480d29373bfcd551e65  tmp-glibc/deploy/ipk/core2-32/xen-libxl-dbg_4.6.4-r0_core2-32.ipk

It is not rebuilt when run again for ndvm
$ MACHINE=xenclient-ndvm ./bb xen-libxl
...
NOTE: Tasks Summary: Attempted 1696 tasks of which 1695 didn't need to be rerun and all succeeded.
$ sha1sum tmp-glibc/deploy/ipk/*/xen-libxl*
474141cbc20325c910679480d29373bfcd551e65  tmp-glibc/deploy/ipk/core2-32/xen-libxl-dbg_4.6.4-r0_core2-32.ipk

Looks good to me.

**Comment by eric-ch** [comment]
TL;DR: +1, LGTM, build pending.

After doing some testing with Bitbake(1.28)/OE(Morty) upstream, it appears that this is not related to the ocaml-findlib class bringing machine specific things to the stamps.
Having a machine-independent recipe depend on a machine-specific one will reproduce the behavior consistently (the machine-independent recipe is rebuilt when the MACHINE variable is changed).
Some things pointed by that could be improved though, that will be discussed on the ticket.

**Comment by eric-ch** [comment]
Build (833)[http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/833]
Includes:
- https://github.com/OpenXT/meta-openxt-ocaml-platform/pull/4
- https://github.com/OpenXT/openxt/pull/243
- https://github.com/OpenXT/xenclient-oe/pull/610
- https://github.com/OpenXT/xenclient-oe/pull/618

**Comment by eric-ch** [comment]
Build [833](http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/833) builds, boots and passes smoke testing (UIVM+NDVM, create win10x64 guest with stubdom, install tools).
From the build output, recipes built more than once:
```      2 initramfs-xenclient-1.0-r2
      2 libdrm-2.4.65-r0
      2 networkmanager-0.9.2.0-r5.1
      2 networkmanager-certs-1.0-r0
      2 openssh-7.1p2-r0.5
      2 refpolicy-mcs-2.20141203-r0.3
      2 xenclient-toolstack-0+gitAUTOINC+54e476cc75-r0
      2 xenfb2-0+gitAUTOINC+8417360fca-r0
      3 init-ifupdown-1.0-r7.1
      3 iproute2-4.1.1-r0.2
      3 modules-1.0-r1
      3 opkg-arch-config-1.0-r1
      3 rsyslog-7.6.1-r0
      3 sysvinit-2.86-r61
      3 udev-182-r9.4.1
      3 xenclient-feed-configs-833-r15
      4 base-files-3.0.14-r92xc10
      4 depmodwrapper-cross-1.0-r0
      4 initscripts-1.0-r155.1
      4 linux-openxt-4.9.20-r0
      4 shadow-securetty-4.2.1-r3
      4 v4v-module-gitAUTOINC+28423277b9-r0```
